pages/api/coupon-fraud/claim.js

```diff
@@ -0,0 +1,78 @@
+import { getForbiddenReponse, getOkReponse } from '../../../server/server';
+import { Op } from 'sequelize';
+import { couponEndpoint } from '../../../server/coupon-fraud/coupon-endpoint';
+import { CouponClaim, CouponCode } from '../../../server/coupon-fraud/database';
+
+async function checkVisitorClaimedRecently(visitorId) {
+  const oneHourBefore = new Date();
+  oneHourBefore.setHours(oneHourBefore.getHours() - 1);
+
+  return await CouponClaim.findOne({
+    where: {
+      visitorId,
+      timestamp: {
+        [Op.between]: [oneHourBefore, new Date()],
+      },
+    },
+  });
+}
+
+async function getVisitorClaim(visitorId, couponCode) {
+  return await CouponClaim.findOne({
+    where: { visitorId, couponCode },
+  });
+}
+
+/**
+ * Checks if given visitor has already used a coupon code.
+ */
+export async function checkCoupon(code) {
+  return await CouponCode.findOne({
+    where: {
+      code: {
+        [Op.eq]: code,
+      },
+    },
+  });
+}
+
+/**
+ * Saves coupon code into database. If it already exists, we update its timestamp.
+ */
+export async function saveCouponCode(visitorId, couponCode) {
```

Comment:
This function does not save or store a coupon. It applies a coupon (or creates a claim). I'd rename it and also improved the comment accordingly.